cooltools/lib/common.py

```diff
@@ -9,6 +9,76 @@
 import bioframe
 
 
+def assign_view_paired(
+    features,
+    view_df,
+    cols_paired=["chrom1", "start1", "end1", "chrom2", "start2", "end2"],
+    cols_view=["chrom", "start", "end"],
+    features_view_cols=["region1", "region2"],
+    view_name_col="name",
+    drop_unassigned=False,
+):
+    """Assign region names from the view to each feature
+    
+    Assigns a regular 1D view independently to each side of a bedpe-style dataframe.
+    Will add two columns with region names (`features_view_cols`)
+
+    Parameters
+    ----------
+    features : pd.DataFrame
+        bedpe-style dataframe
+    view_df : pandas.DataFrame
+        ViewFrame specifying region start and ends for assignment. Attempts to
+        convert dictionary and pd.Series formats to viewFrames.
+    cols_paired : list of str
+        he names of columns containing the chromosome, start and end of the
+        genomic intervals. The default values are 'chrom', 'start', 'end'.
+    cols_view : list of str
+        The names of columns containing the chromosome, start and end of the
+        genomic intervals in the view. The default values are 'chrom', 'start', 'end'.
+    features_view_cols : list of str
+        Names of the columns where to save the assigned region names
+    view_name_col : str
+        Column of ``view_df`` with region names. Default 'name'.
+    drop_unassigned : bool
+        If True, drop intervals in df that do not overlap a region in the view.
+        Default False.
+    """
+    features = features.copy()
+    features.reset_index(inplace=True, drop=True)
+
+    cols_left = cols_paired[:3]
+    cols_right = cols_paired[3:]
+
+    bioframe.core.checks.is_bedframe(features, raise_errors=True, cols=cols_left)
+    bioframe.core.checks.is_bedframe(features, raise_errors=True, cols=cols_right)
+    view_df = bioframe.core.construction.make_viewframe(
+        view_df, view_name_col=view_name_col, cols=cols_view
+    )
+    features = bioframe.assign_view(
+        features,
+        view_df,
+        drop_unassigned=drop_unassigned,
+        df_view_col=features_view_cols[0],
+        view_name_col=view_name_col,
+        cols=cols_left,
+        cols_view=cols_view,
+    )
+    features[cols_right[1:]] = features[cols_right[1:]].astype(
+        int
```

Comment:
can you check if this cast is needed for the final output - i.e. for after the second `assign_view` application ? just in case (also is it an expected behavior on bioframe side of things ? issue-worthy or not ?)

Reply:
This is because the df start off with numpy int dtypes, and then bioframe converts the other columns to numpy float64. But if it is pd.Int64Dtype from the beginning, this cast is not necessary.
In the second run of assign_view the cols_right[1:] get cast to pd.Int64Dtype, and the other ones are already that from this first cast... It's confusing. But the second cast is not necessary.
I don't know if this should be an issue in bioframe, I remember extended discussions about dtypes there and I think we couldn't find a way to avoid this sort of weird casting...